src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -420,6 +551,55 @@ private ODataNestedResourceInfo ReadNestedResourceInfo()
             return link;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeletedResource"/>
+        /// </summary>
+        /// <returns>An <see cref="MaterializerDeletedEntry"/> of the read <see cref="ODataDeletedResource"/>.</returns>
+        private MaterializerDeletedEntry ReadDeletedResource()
+        {
+            this.ExpectState(ODataReaderState.DeletedResourceStart);
+
+            ODataDeletedResource result = (ODataDeletedResource)this.reader.Item;
+            MaterializerDeletedEntry entry;
+
+            if (result != null)
+            {
+                entry = MaterializerDeletedEntry.CreateDeletedEntry(
+                    result,
+                    this.materializerContext);
+
+                if (this.items?.Count > 0)
+                {
+                    this.items.Push(entry);
+                }
+
+                do
+                {
+                    this.AssertRead();
+
+                    switch (this.reader.State)
+                    {
+                        case ODataReaderState.DeletedResourceEnd:
+                            if (this.items?.Count > 0)

```

Comment:
Is there a scenario where we'd end up here and there's no item in the stack to pop?

Reply:
the stack is only used when we are reading a delta payload. If we are just reading a single payload then this.items will be null hence the check.

Reply:
@ElizabethOkerio Here and other places where you have given me the same response, can those particular methods be called other than when we're reading a delta payload? If the answer is "no", can `this.items` be null when we're reading a delta payload? If the answer is "no", why do we need the null propagation operator?

Reply:
yes we actually do not need it. I'll remove it from some methods that can only be called when we are reading a delta payload.